experimental/packages/opentelemetry-sdk-metrics-base/src/view/AttributesProcessor.ts

```diff
@@ -43,4 +43,24 @@ export class NoopAttributesProcessor extends AttributesProcessor {
   }
 }
 
+/**
+ * {@link AttributesProcessor} that filters by allowed attribute names and drops any names that are not in the
+ * allow list.
+ */
+export class FilteringAttributesProcessor extends AttributesProcessor {
+  constructor(private _allowedAttributeNames: string[]) {
+    super();
+  }
+
+  process(incoming: Attributes, _context:Context): Attributes {
+    const filteredAttributes: Attributes = {};
+    for(const allowedAttributeName of this._allowedAttributeNames){
+      if(allowedAttributeName in incoming){
```

Comment:
I would probably prefer to use something like `Object.hasOwnProperty` in order to avoid any weirdness with the object prototype. You would want  to implement it respecting https://eslint.org/docs/rules/no-prototype-builtins like this maybe:

```suggestion
      if(Object.prototype.hasOwnProperty.call(incoming, allowedAttributeName)){
```

Reply:
I added a [PR](https://github.com/open-telemetry/opentelemetry-js-api/pull/155) to explicitly document that only own-enumerable attribute keys count.

Reply:
I updated this in the most recent commit. 
Should I also make sure that the property is enumerable, or is that something we can assume at the point where attributes are processed? :thinking:

Reply:
Probably we can use `Object.keys` to fetch the own-enumerable keys then filter them with the allowed names list and pick all of them at once. In this way, we also avoid the for-of loop since it is kinda slower than looping over an array since it depends on iterators.

Reply:
Thanks for the feedback. I changed it to use `Object.keys` instead. :slightly_smiling_face: